package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,420 @@
+    def dimensions(self, box):
+        self._unitcell[:] = core.triclinic_vectors(box)
+
+
+class H5MDReader(base.ReaderBase):
+    """Reader for the H5MD format.
+    (See `h5md documentation <https://nongnu.org/h5md/h5md.html#>`_
+    for a detailed overview)
+
+    Currently reads .h5md files with the following HDF5 hierarchy:
+
+    .. code-block:: text
+
+        Notation:
+        (name) is an HDF5 group that the reader recognizes
+        {name} is an HDF5 group with arbitrary name
+        [variable] is an HDF5 dataset
+        <dtype> is dataset datatype
+        +-- is an attribute of a group or dataset
+
+        H5MD root
+         \-- (h5md)
+            +-- version <int>
+            \-- author
+                +-- name <str>, author's name
+                +-- email <str>, optional email address
+            \-- creator
+                +-- name <str>, file that created .h5md file
+                +-- version
+         \-- (particles)
+            \-- {group1}
+                \-- (box)
+                    +-- dimension : <int>, number of spatial dimensions
+                    +-- boundary : <str>, boundary conditions of unit cell
+                    \-- (edges)
+                        \-- [step] <int>, gives frame
+                        \-- [value] <float>, gives box dimensions
+                \-- (position)
+                    \-- [step] <int>, gives frame
+                    \-- [time] <float>, gives time
+                        +-- units <str>
+                    \-- [value] <float>, gives numpy arrary of positions
+                                         with shape (n_atoms, 3)
+                        +-- units <str>
+                \-- (velocity)
+                    \-- [step] <int>, gives frame
+                    \-- [time] <float>, gives time
+                        +-- units <str>
+                    \-- [value] <float>, gives numpy arrary of velocities
+                                         with shape (n_atoms, 3)
+                        +-- units <str>
+                \-- (force)
+                    \-- [step] <int>, gives frame
+                    \-- [time] <float>, gives time
+                        +-- units <str>
+                    \-- [value] <float>, gives numpy arrary of forces
+                                         with shape (n_atoms, 3)
+                        +-- units <str>
+                \-- (data)
+                    \-- (dt)
+                        \-- [step] <int>, gives frame number
+                        \-- [value] <float>, gives dt
+                    \-- (lambda)
+                        \-- [step] <int>, gives frame
+                        \-- [value] <float>, gives lambda
+                    \-- (step)
+                        \-- [step] <int>, gives frame
+                        \-- [value] <int>, gives step
+
+
+    .. note::
+        The reader does not currently read mass data.
+
+    """
+
+    format = 'H5MD'
+    # units are added from h5md file
```

Comment:
Your code _changes_ `units` so `units` cannot be a class variable. If you had two H5MDReader then the second one would suddenly start with the units of the first one and then the second one changes the units of the first one. This would be bad™ (and confusing and lead to really difficult to find errors.)

Instead, add a comment here saying that `units` is defined as an instance-level variable and set from the H5MD file.

Then, in `__init__`, initialize `self.units` exactly as done here. Then use `_translate_h5md_units()` as before.

Reply:
E.g. see https://dzone.com/articles/python-class-attributes-vs-instance-attributes